ts/features/pn/aar/saga/fetchNotificationDataSaga.ts

```diff
@@ -0,0 +1,125 @@
+import { NonEmptyString } from "@pagopa/ts-commons/lib/strings";
+import * as E from "fp-ts/lib/Either";
+import { call, put, select } from "typed-redux-saga/macro";
+import { MessageBodyMarkdown } from "../../../../../definitions/backend/MessageBodyMarkdown";
+import { MessageSubject } from "../../../../../definitions/backend/MessageSubject";
+import { ThirdPartyMessage as PnThirdPartyMessage } from "../../../../../definitions/pn/ThirdPartyMessage";
+import { ThirdPartyMessage as AarThirdPartyMessage } from "../../../../../definitions/pn/aar/ThirdPartyMessage";
+import { pnMessagingServiceIdSelector } from "../../../../store/reducers/backendStatus/remoteConfig";
+import { SessionToken } from "../../../../types/SessionToken";
+import { isTestEnv } from "../../../../utils/environment";
+import { getServiceDetails } from "../../../services/common/saga/getServiceDetails";
+import { profileFiscalCodeSelector } from "../../../settings/common/store/selectors";
+import { SendAARClient } from "../api/client";
+import {
+  populateStoresWithEphemeralAarMessageData,
+  setAarFlowState
+} from "../store/actions";
+import { currentAARFlowData } from "../store/selectors";
+import { AarFlowStates, sendAARFlowStates } from "../utils/stateUtils";
+import { isPnTestEnabledSelector } from "../../../../store/reducers/persistedPreferences";
+
+export function* fetchAarDataSaga(
+  fetchData: SendAARClient["getAARNotification"],
+  sessionToken: SessionToken
+) {
+  const currentState = yield* select(currentAARFlowData);
+  const isTest = yield* select(isPnTestEnabledSelector);
+  if (currentState.type !== sendAARFlowStates.fetchingNotificationData) {
+    return;
+  }
+  try {
+    const result = yield* call(fetchData, {
+      Bearer: `Bearer ${sessionToken}`,
+      iun: currentState.iun,
+      mandateId: currentState.mandateId,
+      "x-pagopa-pn-io-src": "QRCODE",
+      isTest
+    });
+
+    if (E.isLeft(result)) {
+      yield* put(
+        setAarFlowState({
+          type: sendAARFlowStates.ko,
+          previousState: currentState
+        })
+      );
+      return;
+    }
+    const { status, value } = result.right;
+    if (status !== 200) {
+      yield* put(
+        setAarFlowState({
+          type: sendAARFlowStates.ko,
+          previousState: currentState,
+          error: value
+        })
+      );
+      return;
+    }
+
+    const payload = yield* call(
+      aarMessageDataPayloadFromResponse,
+      value,
+      currentState
+    );
+
+    if (payload === undefined) {
+      return;
```

Comment:
Let's temporarily throw an error here:
`throw Error("Unable to compute placeholder payload for AAR")`

We will change it later but in this way, we assure that the case is handled by the `catch` below